webdav/dandidav.py

```diff
@@ -0,0 +1,549 @@
+        try:
+            return self.asset.get_raw_metadata()["encodingFormat"]
+        except KeyError:
+            return "application/octet-stream"
+
+    def get_display_info(self) -> dict:
+        return {"type": "Blob asset"}
+
+    def is_link(self) -> bool:
+        # Fix for <https://github.com/mar10/wsgidav/issues/301>
+        return False
+
+    def get_etag(self) -> str | None:
+        try:
+            return self.asset.get_raw_digest()
+        except NotFoundError:
+            return None
+
+    def support_etag(self) -> bool:
+        return True
+
+    def get_creation_date(self) -> float:
+        return self.asset.created.timestamp()
+
+    def get_last_modified(self) -> float:
+        return self.asset.modified.timestamp()
+
+
+class VersionResource(AssetFolder):
+    """
+    A Dandiset at a specific version, containing top-level assets and asset
+    folders
+    """
+
+    def __init__(self, path: str, environ: dict, dandiset: RemoteDandiset) -> None:
+        super().__init__(path, environ, dandiset, "")
+
+    def get_display_info(self) -> dict[str, str]:
+        return {"type": "Dandiset version"}
+
+    def get_member_list(
+        self,
+    ) -> list[BlobResource | ZarrResource | AssetFolder | DandisetYaml]:
+        members = super().get_member_list()
+        members.append(
+            DandisetYaml(
+                join_uri(self.path, "dandiset.yaml"), self.environ, self.dandiset
+            )
+        )
+        members.sort(key=attrgetter("name"))
+        return members
+
+    def get_member_names(self) -> list[str]:
+        names = super().get_member_names()
+        names.append("dandiset.yaml")
+        names.sort()
+        return names
+
+    def get_member(
+        self, name: str
+    ) -> BlobResource | ZarrResource | AssetFolder | DandisetYaml | None:
+        if name == "dandiset.yaml":
+            return DandisetYaml(
+                join_uri(self.path, "dandiset.yaml"), self.environ, self.dandiset
+            )
+        else:
+            return super().get_member(name)
+
+    def is_link(self) -> bool:
+        # Fix for <https://github.com/mar10/wsgidav/issues/301>
+        return False
+
+    def get_creation_date(self) -> float:
+        return self.dandiset.version.timestamp()
+
+    def get_last_modified(self) -> float:
+        return self.dandiset.version.modified.timestamp()
+
+
+class DandisetYaml(DAVNonCollection):
+    def __init__(self, path: str, environ: dict, dandiset: RemoteDandiset) -> None:
+        super().__init__(path, environ)
+        self.dandiset = dandiset
+
+    def get_content(self) -> IO[bytes]:
+        yaml = YAML(typ="safe")
+        yaml.default_flow_style = False
+        out = io.BytesIO()
+        yaml.dump(self.dandiset.get_raw_metadata(), out)
+        out.seek(0)
+        return out
+
+    def get_content_length(self) -> None:
+        return None
+
+    def get_content_type(self) -> str:
+        return "application/yaml"
```

Comment:
Even though indeed https://www.iana.org/assignments/media-types/media-types.xhtml defines only `application/yaml` there is only a draft https://datatracker.ietf.org/doc/draft-ietf-httpapi-yaml-mediatypes/10/ for that and it says that prior (not registered) ones are `text/yaml` etc.

With `application/` it causes browsers to download the file instead of allowing for viewing it.  That makes interactions in the browser with such files less convenient. May be that is why github when serves it -- says it is `text/plain`, I also get it without any from my apache2 but visualized in the browser.  So for pragmatic reasons, and because here we know it is not data serialization but really a machine and text readable record, and we do know encoding AFAIK, let's do

```suggestion
        return "text/yaml; charset=utf-8"
```